### PR TITLE
Hyodrogen Horizons initial updates

### PR DESCRIPTION
**Pull request by kyle-flynn**
Provides foundational work to enable work on Hydrogen Horizons FGC 2023 game. There is still a good amount of UI work that needs to be done, but this should allow you to start work on the field control system.

### CODE REVIEW

**Review by scarabaeus — APPROVED** [review]
APPROVED -- LOOKS GREAT!

**Comment by NoahAndrews** [comment]
image

Might want to cut down on the all-caps, haha

**Review by NoahAndrews — CHANGES_REQUESTED** [review]

File: lib/models/src/seasons/HydrogenHorizons.ts [hunk 1]
```diff
@@ -0,0 +1,370 @@
...
+  key: 'fgc_2023',
+  program: 'fgc',
+  name: 'Hydrogen Horizons',
+  functions
+};
+
+export interface MatchDetails extends MatchDetailBase {
+  redHydrogenPoints: number;
+  redOxygenPoints: number;
+  redAlignment: number;
```
[the remark this answers is not in the document]
Comment (NoahAndrews) [reply lib/models/src/seasons/HydrogenHorizons.ts:25]: Disregard, added to #69

**Review by NoahAndrews — APPROVED** [review]
Approving this PR for merging as-is. I've made some improvements in #69, but they can be merged separately.

**Pull request merged**